### ISSUE TITLE
E3006 false positive since 1.54.0: four AWS::Serverless::* resource types have no bundled schema (CapacityProvider, WebSocketApi, MicrovmImage, NetworkConnector)

### ISSUE THREAD

**Issue by roger-zhangg** [aws-cloudformation/cfn-lint#4678]
### CloudFormation Lint Version

`cfn-lint 1.56.0` (also reproduced on 1.54.0, 1.55.0, 1.55.1)

### What operating system are you using?

Mac (also Ubuntu 24.04 on GitHub Actions)

### Describe the bug

Since #4491 dropped the `aws-sam-translator` dependency and moved `AWS::Serverless::*` validation onto bundled schemas, `cfnlint/data/schemas/providers/sam.json` covers 9 of the 13 SAM resource types the translator supports. E3006 now fires on the other four, so valid SAM templates fail to lint:

| Resource type | SAM docs | Added to `aws-sam-translator` |
|---|---|---|
| `AWS::Serverless::CapacityProvider` | [sam-resource-capacityprovider](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-capacityprovider.html) | 2025-12-01 (Lambda Managed Instances); present in v1.104.0 |
| `AWS::Serverless::WebSocketApi` | [sam-resource-websocketapi](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-websocketapi.html) | 2026-04-08 |
| `AWS::Serverless::MicrovmImage` | [sam-resource-microvmimage](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-microvmimage.html) | 2026-07-01 |
| `AWS::Serverless::NetworkConnector` | [sam-resource-networkconnector](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-networkconnector.html) | 2026-07-01 |

Each has a `samtranslator/internal/schema_source/aws_serverless_*.py` module and definitions in the published SAM schema, so they are ordinary released SAM resource types rather than anything preview-only.

The gap is visible directly:

```console
$ python -c "from cfnlint.schema.manager import PROVIDER_SCHEMA_MANAGER as m; print(sorted(t for t in m.get_resource_types(region='us-east-1') if t.startswith('AWS::Serverless::')))"
['AWS::Serverless::Api', 'AWS::Serverless::Application', 'AWS::Serverless::Connector', 'AWS::Serverless::Function', 'AWS::Serverless::GraphQLApi', 'AWS::Serverless::HttpApi', 'AWS::Serverless::LayerVersion', 'AWS::Serverless::SimpleTable', 'AWS::Serverless::StateMachine']
```

`test/fixtures/schemas/providers/sam.json` on `main` (f4c72e0, after v1.56.0) still lists the same nine, so this is not fixed on `main` either.

Version bisect with the reproduction template below:

| cfn-lint | Result |
|---|---|
| 1.51.3, 1.52.1, 1.53.0, 1.53.3 | exit 0, no findings |
| 1.54.0, 1.55.0, 1.55.1, 1.56.0 | exit 2, `E3006` |

#4607 tracked the rule-level fallout from #4491 and is complete; missing resource types look like a separate gap, and I could not find an existing issue or PR for them.

### Expected behavior

`AWS::Serverless::CapacityProvider`, `AWS::Serverless::WebSocketApi`, `AWS::Serverless::MicrovmImage` and `AWS::Serverless::NetworkConnector` are recognised resource types, as they were up to 1.53.3, so E3006 does not fire on templates that use them.

More generally: it would help downstream consumers if the set of SAM types in `providers/sam.json` tracked `aws-sam-translator`, since a type added there now becomes a lint failure until cfn-lint catches up.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Transform: AWS::Serverless-2016-10-31
Resources:
  CapacityProvider:
    Type: AWS::Serverless::CapacityProvider
    Properties:
      VpcConfig:
        SubnetIds: [subnet-0123456789abcdef0]
        SecurityGroupIds: [sg-0123456789abcdef0]
      ScalingConfig:
        MaxVCpuCount: 20
        AverageCPUUtilization: 70.0
```

```console
$ cfn-lint template.yaml --region us-east-1
E3006 Resource type 'AWS::Serverless::CapacityProvider' does not exist in 'us-east-1'
template.yaml:5:5
```

The other three behave the same way; swapping in a bare `Type: AWS::Serverless::WebSocketApi` (or `MicrovmImage`, or `NetworkConnector`) yields E3006 alongside the expected missing-required-property findings.

### Impact

AWS SAM CLI takes cfn-lint as a dependency for `sam validate --lint`. Picking up 1.55.1 shipped this in SAM CLI 1.166.0, where it broke CI for repositories whose templates use these types — for example [aws/aws-lambda-web-adapter](https://github.com/aws/aws-lambda-web-adapter/actions/runs/33804109734/job/100813179802), whose `examples/fastapi-response-streaming-lmi/template.yaml` went from passing on SAM CLI 1.165.0 (cfn-lint 1.52.1) to failing on 1.166.0. SAM CLI has pinned back to `cfn-lint<1.54` in the meantime.

Happy to open a PR adding the four schemas if you can point me at how `providers/sam.json` and the SAM provider schemas are generated — they are gitignored, so I could not tell from the repo which step produces them.

**Comment by roger-zhangg** [comment]
https://github.com/aws/aws-sam-cli/issues/9248
There's another report on E3724 false positive starting cfn-lint 1.154.0. Please help to take a look. Thanks
